### DISCUSSION TITLE
tabview lazy loading

### DISCUSSION THREAD

**Discussion by ymajoros** [primefaces/community#567] [PrimeNG]
Is tabview lazy loading still supported in v15+? 

It seems to be missing from the documentation, though a quick look in the code seems there is still something for that. Is that just a documentation bug, or is the feature not supported for some reason?

**Comment by mehmetcetin01140** [comment]
Hi,

Tabview does not support lazy load.

**Reply by ymajoros** [reply]
Are you sure? it really seems to work though. It used to be documented until the documentation refactoring, and the code didn't seem to change in that aspect.

**Reply by riker09** [reply]
TabView *does* support lazy loading. The documentation was there last with PrimeNG 14: https://www.primefaces.org/primeng-v14-lts/tabview

Basically all you have to do is wrap the content in an `<ng-template pTemplate="content"></ng-template>` and you're done.

Long example from linked v14 documentation:

> Lazy loading helps initial load performance by only initializing the active tab, inactive tabs are not initialized until they get selected. A lazy loaded tabpanel contents are cached by default so that upon reselection, they are not created again. You may use cache property on TabPanel to configure this behavior. A TabPanel is specified as lazy when there is a ngTemplate with pTemplate="content" in it.

```html
<p-tabView>
    <p-tabPanel header="Header 1">
        Content 1
    </p-tabPanel>
    <p-tabPanel header="Header 2">
        <ng-template pTemplate="content">
            Complex Content to Lazy Load
        </ng-template>
    </p-tabPanel>
    <p-tabPanel header="Header 3">
        <ng-template pTemplate="content">
            Complex Content to Lazy Load
        </ng-template>
    </p-tabPanel>
</p-tabView>
``` 

I have created a ticket for this before I stumbled accross this page: https://github.com/primefaces/primeng/issues/14305

**Reply by carcigenicate** [reply]
I can confirm that `<ng-template pTemplate="content">` does cause lazy loading (Angular/PrimeNG 16). If I put a `{{ console.log(...) }}` inside the `p-tabPanel`, I can see that it only loads the tabs that have been opened.

Without the `content` template, all tabs fire their `console.log`s immediately on page load. With the `content` template though, only the opened tabs fire logs. All tabs that have been *previously* opened though also fire on every tab change, so if the user goes through all the tabs, it looks like they all get re-rendered on every tab change. If I do `[cache]=false` on the `p-tabPanel` though, I only get logs for the tab that was opened.

CD firing might be complicating the printouts since they're in the template, but I found that using the `content` template paired with, counter-intunitively, `[cache]=false` gave the best performance.
